presto-verifier/src/main/java/com/facebook/presto/verifier/resolver/StructuredColumnMismatchResolver.java

```diff
@@ -68,22 +68,35 @@ else if (columnType instanceof MapType) {
                 MapColumnChecksum controlChecksum = (MapColumnChecksum) mismatchedColumn.getControlChecksum();
                 MapColumnChecksum testChecksum = (MapColumnChecksum) mismatchedColumn.getTestChecksum();
 
+                // Cardinality mismatch. Do not resolve.
                 if (!isCardinalityMatched(controlChecksum, testChecksum)) {
                     return Optional.empty();
                 }
 
                 boolean keyContainsFloatingPoint = containsFloatingPointType(((MapType) columnType).getKeyType());
                 boolean valueContainsFloatingPoint = containsFloatingPointType(((MapType) columnType).getValueType());
+                // No pure floating point types. Do not resolve.
                 if (!keyContainsFloatingPoint && !valueContainsFloatingPoint) {
                     return Optional.empty();
                 }
+                // Not-floating point keys mismatch. Do not resolve.
                 if (!keyContainsFloatingPoint &&
                         !Objects.equals(controlChecksum.getKeysChecksum(), testChecksum.getKeysChecksum())) {
                     return Optional.empty();
                 }
-                if (!valueContainsFloatingPoint &&
-                        !Objects.equals(controlChecksum.getValuesChecksum(), testChecksum.getValuesChecksum())) {
-                    return Optional.empty();
+                // Not-floating point values.
+                if (!valueContainsFloatingPoint) {
+                    // If we got values checksums and they are not matching, then do not resolve.
+                    if (!Objects.equals(controlChecksum.getValuesChecksum(), testChecksum.getValuesChecksum())) {
+                        return Optional.empty();
+                    }
+
+                    // Values checksums either match or missing. Checking the whole checksum if the values checksums are missing.
```

Comment:
why would it be missing?

Reply:
@rschlussel 
When map is empty or even null completely?
When no rows returned?
Such corner cases.